### PR TITLE
1D tests for plasma acceleration

### PR DESCRIPTION
**Pull request by prkkumar**
This PR adds a 1D Python test for plasma wakefield acceleration and a 1D C++ test for laser-plasma acceleration

- [x] depends on #2582

### CODE REVIEW

**Comment by ax3l** [comment]
@prkkumar ready for rebase (#2582 was merged :) )

File: Examples/Physics_applications/plasma_acceleration/PICMI_inputs_plasma_acceleration_1d.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                             lower_bound = [zmin],
+                             upper_bound = [zmax],
+                             lower_boundary_conditions = ['open'],
+                             upper_boundary_conditions = ['open'],
+                             lower_boundary_conditions_particles = ['absorbing'],
+                             upper_boundary_conditions_particles = ['absorbing'],
+                             moving_window_velocity = moving_window_velocity,
+                             warpx_max_grid_size=32)
+
+solver = picmi.ElectromagneticSolver(grid=grid, cfl=0.9)
```
Comment (ax3l) [inline Examples/Physics_applications/plasma_acceleration/PICMI_inputs_plasma_acceleration_1d.py]:
```suggestion
solver = picmi.ElectromagneticSolver(grid=grid, cfl=0.999)
```

File: Regression/WarpX-tests.ini [hunk 2]
```diff
@@ -1568,6 +1568,25 @@ particleTypes = beam
 analysisRoutine = Examples/analysis_default_regression.py
 tolerance = 1.e-14
 
+[Python_PlasmaAcceleration_1d]
+buildDir = .
+inputFile = Examples/Physics_applications/plasma_acceleration/PICMI_inputs_plasma_acceleration_1d.py
+runtime_params =
+customRunCmd = python PICMI_inputs_plasma_acceleration_1d.py
+dim = 1
+addToCompileString = USE_PYTHON_MAIN=TRUE PYINSTALLOPTIONS="--user --prefix=" USE_PSATD=FALSE
```
Comment (ax3l) [inline Regression/WarpX-tests.ini]:
```suggestion
addToCompileString = USE_PYTHON_MAIN=TRUE PYINSTALLOPTIONS="--user --prefix="
```

discussed: this is using PICMI w/o 1D classes, that's why it crashes

File: requirements.txt [hunk 3]
```diff
@@ -6,7 +6,6 @@ periodictable~=1.5
 picmistandard==0.0.16
```
Comment (RemiLehe) [inline requirements.txt]: The PR that introduces the 1D Cartesian grid was merged in `picmistandard`.
@ax3l Do I understand correctly that, for the tests to pass here, we should do a new release of `picmistandard` (0.0.17) and update the above requirement?

**Review by RemiLehe — DISMISSED** [review]

File: Python/pywarpx/picmi.py [hunk 4]
```diff
@@ -463,6 +463,57 @@ def initialize_inputs(self):
...
+        self.max_grid_size_x = kw.pop('warpx_max_grid_size_x', None)
+        self.blocking_factor = kw.pop('warpx_blocking_factor', None)
+        self.blocking_factor_x = kw.pop('warpx_blocking_factor_x', None)
+
+        self.potential_xmin = None
+        self.potential_xmax = None
+        self.potential_ymin = None
+        self.potential_ymax = None
+        self.potential_zmin = kw.pop('warpx_potential_lo_z', None)
+        self.potential_zmax = kw.pop('warpx_potential_hi_z', None)
```
Comment (roelof-groenewald) [inline Python/pywarpx/picmi.py:478]: On lines 405 & 406 of `ElectrostaticSolver.cpp` the `potential_xlo/hi` values are used to set the 0 index values of `phi_bc_values_lo/hi` which is used to set the boundary potential if Dirichlet BCs are used. I think to keep things consistent those should be changed to use the z-potentials as well.

**Review by roelof-groenewald — CHANGES_REQUESTED** [review]

**Comment by ax3l** [comment]
@RemiLehe @prkkumar can we also add a 1D "C++" test `inputs_1d` in https://github.com/ECP-WarpX/WarpX/tree/development/Examples/Physics_applications/laser_acceleration ? I just realized I use this for smoke & acceptance tests in all package managers :)

File: requirements.txt (hunk 3 above)
Comment (ax3l) [reply requirements.txt]:
```suggestion
picmistandard==0.0.17
```

File: requirements.txt [hunk 5]
```diff
@@ -3,10 +3,9 @@ numpy~=1.15
 periodictable~=1.5
 
 # PICMI
-picmistandard==0.0.16
+picmistandard==0.0.17
 #   for development against an unreleased PICMI version, use:
 #picmistandard @ git+https://github.com/picmi-standard/picmi.git#subdirectory=PICMI_Python
```
Comment (ax3l) [inline requirements.txt:8]: Keep empty newline
```suggestion
#picmistandard @ git+https://github.com/picmi-standard/picmi.git#subdirectory=PICMI_Python

```

**Review by ax3l — APPROVED** [review]
Thanks a lot! :tada:

**Pull request merged**